cmd/clusterctl/pkg/client/cluster/objectgraph_test.go

```diff
@@ -849,47 +1146,166 @@ func Test_objectGraph_setSoftDependents(t *testing.T) {
+						WithMachineSets(
+							test.NewFakeMachineSet("cluster1-ms1").
+								WithInfrastructureTemplate(sharedInfrastructureTemplate).
+								WithMachines(
+									test.NewFakeMachine("cluster1-m1"),
+								),
+						).Objs()...)
+
+					objs = append(objs, test.NewFakeCluster("ns1", "cluster2").
+						WithMachineSets(
+							test.NewFakeMachineSet("cluster2-ms1").
+								WithInfrastructureTemplate(sharedInfrastructureTemplate).
+								WithMachines(
+									test.NewFakeMachine("cluster2-m1"),
+								),
+						).Objs()...)
+
+					return objs
+				}(),
+			},
+			wantClusters: map[string][]string{ // wantClusters is a map[Cluster.UID] --> list of UIDs
+				"cluster.x-k8s.io/v1alpha3, Kind=Cluster, ns1/cluster1": {
+					"infrastructure.cluster.x-k8s.io/v1alpha3, Kind=DummyInfrastructureMachineTemplate, ns1/shared", // the shared object should be in both lists
+					"cluster.x-k8s.io/v1alpha3, Kind=Cluster, ns1/cluster1",                                         // the cluster should be tenant of itself
+					"infrastructure.cluster.x-k8s.io/v1alpha3, Kind=DummyInfrastructureCluster, ns1/cluster1",
+					"/v1, Kind=Secret, ns1/cluster1-ca", // the ca secret is a soft owned
+					"/v1, Kind=Secret, ns1/cluster1-kubeconfig",
+					"cluster.x-k8s.io/v1alpha3, Kind=MachineSet, ns1/cluster1-ms1",
+					"bootstrap.cluster.x-k8s.io/v1alpha3, Kind=DummyBootstrapConfigTemplate, ns1/cluster1-ms1",
+					"cluster.x-k8s.io/v1alpha3, Kind=Machine, ns1/cluster1-m1",
+					"infrastructure.cluster.x-k8s.io/v1alpha3, Kind=DummyInfrastructureMachine, ns1/cluster1-m1",
+					"bootstrap.cluster.x-k8s.io/v1alpha3, Kind=DummyBootstrapConfig, ns1/cluster1-m1",
+					"/v1, Kind=Secret, ns1/cluster1-m1",
+				},
+				"cluster.x-k8s.io/v1alpha3, Kind=Cluster, ns1/cluster2": {
+					"infrastructure.cluster.x-k8s.io/v1alpha3, Kind=DummyInfrastructureMachineTemplate, ns1/shared", // the shared object should be in both lists
+					"cluster.x-k8s.io/v1alpha3, Kind=Cluster, ns1/cluster2",                                         // the cluster should be tenant of itself
+					"infrastructure.cluster.x-k8s.io/v1alpha3, Kind=DummyInfrastructureCluster, ns1/cluster2",
+					"/v1, Kind=Secret, ns1/cluster2-ca", // the ca secret is a soft owned
+					"/v1, Kind=Secret, ns1/cluster2-kubeconfig",
+					"cluster.x-k8s.io/v1alpha3, Kind=MachineSet, ns1/cluster2-ms1",
+					"bootstrap.cluster.x-k8s.io/v1alpha3, Kind=DummyBootstrapConfigTemplate, ns1/cluster2-ms1",
+					"cluster.x-k8s.io/v1alpha3, Kind=Machine, ns1/cluster2-m1",
+					"infrastructure.cluster.x-k8s.io/v1alpha3, Kind=DummyInfrastructureMachine, ns1/cluster2-m1",
+					"bootstrap.cluster.x-k8s.io/v1alpha3, Kind=DummyBootstrapConfig, ns1/cluster2-m1",
+					"/v1, Kind=Secret, ns1/cluster2-m1",
 				},
 			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			graph, err := getDetachedObjectGraphWihObjs(tt.fields.objs)
+			gb, err := getDetachedObjectGraphWihObjs(tt.fields.objs)
```

Comment:
Why this change?

Reply:
I'm getting a little bit obsessed by code readability 🤣 
If you prefer I can rollback